From: SJ Park <sj@kernel.org>
Date: Sat, 8 Aug 2026 10:02:33 -0700
Subject: [PATCH] ==== refactor damon_call for unstarted ctx ====

mm/damon: allow NULL or unstarted damon_ctx parameter for damon_call()

Callers of damon_call() should validate the damon_ctx object parameter.
If it is NULL or never damon_start()-ed object, damon_call() could
dereference the NULL pointer or indefinitely hang.  Ensuring all callers
doing the validation correctly has turned out to be difficult.  Handle
the corner cases inside core layer and remove callers' validations.

Patches 1 and 2 respectively allow passing NULL and not yet
damon_start()-ed ctx parameter to damon_call().  Patches 3 and 4 remove
the callers side validations in DMON_RECLAIM and DASMON_LRU_SORT,
respectively.

Signed-off-by: SJ Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_p2zgg0vk | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_p2zgg0vk

diff --git a/hkml_cv_bogus/hkml_cv_bogus_p2zgg0vk b/hkml_cv_bogus/hkml_cv_bogus_p2zgg0vk
new file mode 100644
index 0000000000000..e69de29bb2d1d
-- 
2.47.3

